symbolic-sourcemapcache/src/sourcemapcache/writer.rs

```diff
@@ -267,6 +263,28 @@ impl SourceMapCacheWriter {
 
         Ok(())
     }
+
+    /// Compute line offsets for a source file.
+    ///
+    /// There is always one line offset at the start of the file (even if the file is empty)
+    /// and then another one after every newline (even if the file ends on a newline).
+    pub(crate) fn line_offsets(source: &str) -> impl Iterator<Item = raw::LineOffset> + '_ {
```

Comment:
Maybe one way to avoid this chaining of iterators is to take an "out-parameter":
`&mut Vec<raw::LineOffset>` and just push/extend stuff into it.

I think using if + push is a lot more readable and clear than using `bool::then` + the fact that `Option` is an iterator :-D

Reply:
Thanks, I rewrote it with an out parameter.